addons/survey/models/survey_user_input.py

```diff
@@ -610,6 +611,53 @@ def _get_print_questions(self):
             inactive_questions = self._get_inactive_conditional_questions()
         return survey.question_ids - inactive_questions
 
+    def _get_next_skipped_page_or_question(self):
+        """Get next skipped question or page in case the option 'can_go_back' is set on the survey
+        It loops to the first skipped question or page if 'last_displayed_page_id' is the last
+        skipped question or page."""
+        self.ensure_one()
+        skipped_mandatory_answer_ids = self.user_input_line_ids.filtered(
+            lambda answer: answer.skipped and answer.question_id.constr_mandatory)
+
+        if not skipped_mandatory_answer_ids:
+            return self.env['survey.question']
+
+        page_or_question_key = 'page_id' if self.survey_id.questions_layout == 'page_per_section' else 'question_id'
+        page_or_question_ids = skipped_mandatory_answer_ids.mapped(page_or_question_key).sorted()
+
+        if self.last_displayed_page_id not in page_or_question_ids\
+            or self.last_displayed_page_id == page_or_question_ids[-1]:
+            return page_or_question_ids[0]
+
+        current_page_index = page_or_question_ids.ids.index(self.last_displayed_page_id.id)
+        return page_or_question_ids[current_page_index + 1]
+
+    def _get_skipped_questions(self):
+        self.ensure_one()
+
+        return self.user_input_line_ids.filtered(
+            lambda answer: answer.skipped and answer.question_id.constr_mandatory).question_id
+
+    def _is_last_skipped_page_or_question(self, page_or_question):
```

Comment:
Actually, this has become `is_last_page_or_question` , hasn't it?

Reply:
Well no, there could still be a question or more after a skipped question.

Reply:
`has_skipped_page_or_question_afterwards`?
@tde-banana-odoo please help us sort this out :) I'd like to call it "_show_submit_button" and move it to the controller ^^  But it is not very different from other methods like this one that are present on the model so...